File: baselibrary/src/main/java/com/aslan/baselibrary/base/VBBaseFragment.java
package com.aslan.baselibrary.base;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.viewbinding.ViewBinding;
import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;
import java.lang.reflect.ParameterizedType;

/**
 * 基础类
 *
 * @author Aslan
 * @date 2018/4/11
 */
public abstract class VBBaseFragment<VB extends ViewBinding> extends BaseFragment {

  protected VB mViewBinding;

  @Override
  public View setCusContentView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container) {
    ParameterizedType type = (ParameterizedType) getClass().getGenericSuperclass();
    Class cls = (Class) type.getActualTypeArguments()[0];
    try {
      Method inflate = cls
          .getDeclaredMethod("inflate", LayoutInflater.class, ViewGroup.class, boolean.class);
      mViewBinding = (VB) inflate.invoke(null, inflater, container, false);
    } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
      e.printStackTrace();
    }
    return mViewBinding.getRoot();
  }

  @Override
  public final void iniView(View view) {
    iniView();
  }

  public abstract void iniView();

  @Override
  public void onDestroyView() {
    super.onDestroyView();
    mViewBinding = null;
  }
}

File: baselibrary/src/main/java/com/aslan/baselibrary/base/VBBaseFragment.kt
package com.aslan.baselibrary.base

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.CallSuper
import androidx.viewbinding.ViewBinding
import com.aslan.baselibrary.utils.InflateFragment

/**
 * 基础类
 *
 * @author Aslan
 * @date 2018/4/11
 */
open abstract class VBBaseFragment<VB : ViewBinding>(private val inflate: InflateFragment<VB>) :
    BaseFragment() {
    private var _binding: VB? = null
    protected val mViewBinding get() = _binding!!

    final override fun getLayoutId() = 0

    override fun setCusContentView(inflater: LayoutInflater, container: ViewGroup?): View {
        _binding = inflate.invoke(inflater, container, false)
        return mViewBinding.root
    }

    @CallSuper
    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}